packages/calcite-design-tokens/src/build/preprocessors/store-same-value-theme-tokens.ts

```diff
@@ -0,0 +1,29 @@
+import StyleDictionary from "style-dictionary";
+import { DesignToken } from "style-dictionary/types";
+import { dark, light } from "../dictionaries/index.js";
+import { isThemed } from "../utils/token-types.js";
+import { state } from "../shared/state.js";
+
+export function registerPreprocessorStoreSameValueThemeTokens(): void {
+  StyleDictionary.registerPreprocessor({
+    name: PreprocessorStoreSameValueThemeTokens,
+    preprocessor: async (dictionary) => {
+      const sameValueThemeTokens = light.allTokens
+        .filter((token) => isThemed(token))
+        .filter((token, index) => token.value === dark.allTokens[index].value);
+      const keyToToken = new Map<string, DesignToken>();
+
+      sameValueThemeTokens.forEach((token) => {
+        if (token.key) {
+          keyToToken.set(token.key, token);
+        }
+      });
```

Comment:
```suggestion
      const keyToToken = new Map<string, DesignToken>();

      const sameValueThemeTokens = light.allTokens
        .filter((token, index) => isThemed(token) && token.value === dark.allTokens[index].value)
        .filter((token, index) => token.value === dark.allTokens[index].value)
        .forEach((token) => {
          if (token.key) {
            keyToToken.set(token.key, token);
          }
        });
```

I think this could be simplified, can you confirm?

Reply:
Nice! I'll add that to https://github.com/Esri/calcite-design-system/pull/11860.